File: annotation-processor/src/test/java/me/bristermitten/mittenlib/annotations/exception/ConfigProcessingExceptionTest.java
package me.bristermitten.mittenlib.annotations.exception;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertSame;

import org.junit.jupiter.api.Test;

class ConfigProcessingExceptionTest {

    @Test
    void testConstructor() {
        Throwable cause = new RuntimeException("root cause");
        ConfigProcessingException exception = new ConfigProcessingException("error occurred", cause);

        assertEquals("error occurred", exception.getMessage());
        assertSame(cause, exception.getCause());
    }
}

File: annotation-processor/src/test/scala/me/bristermitten/mittenlib/annotations/exception/ConfigProcessingExceptionTest.scala
package me.bristermitten.mittenlib.annotations.exception

class ConfigProcessingExceptionTest extends munit.FunSuite:

  test("constructor") {
    val cause = new RuntimeException("root cause")
    val exception = new ConfigProcessingException("error occurred", cause)

    assertEquals(exception.getMessage, "error occurred")
    assert(exception.getCause eq cause)
  }
